### PR TITLE
[ADF-3794] Update individual rows without reloading DocumentList

### PR DESCRIPTION
**Pull request by DenysVuika**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

- sync cell text with the node updates

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by eromano** [comment]
there are some e2e that are failing I guess related to this modify

File: lib/content-services/document-list/components/library-name-column/library-name-column.component.ts [hunk 1]
```diff
@@ -16,75 +16,105 @@
...
 
-  makeLibraryTooltip(library: any): string {
-    const { description, title } = library;
+        return description || title || '';
+    }
 
-    return description || title || '';
-  }
+    makeLibraryTitle(library: any, rows: Array<ShareDataRow>): string {
+        const entries = rows.map((r: ShareDataRow) => r.node.entry);
```
Comment (eromano) [inline lib/content-services/document-list/components/library-name-column/library-name-column.component.ts]: can you please avoid to use one char variable name?

File: lib/content-services/document-list/components/library-role-column/library-role-column.component.ts [hunk 2]
```diff
@@ -15,44 +15,84 @@
...
+            }
+        });
+    }
+
+    protected updateValue() {
+        const node = this.context.row.node;
+        if (node && node.entry) {
+            const role: string = node.entry.role;
+            switch (role) {
+                case 'SiteManager':
```
Comment (eromano) [inline lib/content-services/document-list/components/library-role-column/library-role-column.component.ts]: can we create an enum ?   we already have permissions.enum can we create permissions-site.enum ?

File: lib/content-services/document-list/components/library-status-column/library-status-column.component.ts [hunk 3]
```diff
@@ -15,42 +15,73 @@
...
+            }
+        });
+    }
+
+    protected updateValue() {
+        const node = this.context.row.node;
+        if (node && node.entry) {
+            const visibility: string = node.entry.visibility;
+
+            switch (visibility.toUpperCase()) {
```
Comment (eromano) [inline lib/content-services/document-list/components/library-status-column/library-status-column.component.ts]: can we create an enum?

File: lib/content-services/document-list/components/name-column/name-column.component.ts [hunk 4]
```diff
@@ -16,50 +16,81 @@
...
+    node: NodeEntry;
 
-  constructor(private element: ElementRef) {}
+    private sub: Subscription;
 
-  ngOnInit() {
-    this.node = this.context.row.node;
-    if (this.node && this.node.entry) {
-      this.displayText = this.node.entry.name || this.node.entry.id;
+    constructor(private element: ElementRef, private api: AlfrescoApiService) {}
```
Comment (eromano) [inline lib/content-services/document-list/components/name-column/name-column.component.ts]: would it better as we discusses rename api to alfrescoApiService?

File: lib/core/datatable/components/datatable/datatable-cell.component.ts [hunk 5]
```diff
@@ -41,20 +53,46 @@ export class DataTableCellComponent implements OnInit {
...
-    @Input()
-    value: any;
+    value$ = new BehaviorSubject<any>('');
 
     @Input()
     tooltip: string;
 
+    private sub: Subscription;
+
+    constructor(protected api: AlfrescoApiService) {}
```
Comment (eromano) [inline lib/core/datatable/components/datatable/datatable-cell.component.ts]: would it better as we discuss rename api to alfrescoApiService?

File: lib/core/datatable/components/datatable/date-cell.component.ts [hunk 6]
```diff
@@ -50,8 +51,8 @@ export class DateCellComponent extends DataTableCellComponent {
         return 'medium';
     }
 
-    constructor(userPreferenceService: UserPreferencesService) {
-        super();
+    constructor(userPreferenceService: UserPreferencesService, apiService: AlfrescoApiService) {
```
Comment (eromano) [inline lib/core/datatable/components/datatable/date-cell.component.ts]: would it better as we discuss rename apiService to alfrescoApiService? it helps consistency

File: lib/core/datatable/components/datatable/filesize-cell.component.ts [hunk 7]
```diff
@@ -17,15 +17,20 @@
...
-            <span [title]="tooltip">{{ value | adfFileSize }}</span>
+            <span [title]="tooltip">{{ (value$ | async) | adfFileSize }}</span>
         </ng-container>
     `,
     encapsulation: ViewEncapsulation.None,
     host: { class: 'adf-filesize-cell' }
 })
-export class FileSizeCellComponent extends DataTableCellComponent {}
+export class FileSizeCellComponent extends DataTableCellComponent {
+    constructor(apiService: AlfrescoApiService) {
```
Comment (eromano) [inline lib/core/datatable/components/datatable/filesize-cell.component.ts]: would it better as we discuss rename apiService to alfrescoApiService? it helps consistency

File: lib/core/datatable/components/datatable/location-cell.component.ts [hunk 8]
```diff
@@ -15,46 +15,56 @@
...
-export class LocationCellComponent extends DataTableCellComponent implements OnInit {
-
+export class LocationCellComponent extends DataTableCellComponent
+    implements OnInit {
     @Input()
     link: any[];
 
-    @Input()
-    displayText: string = '';
+    constructor(apiService: AlfrescoApiService) {
```
Comment (eromano) [inline lib/core/datatable/components/datatable/location-cell.component.ts]: would it better as we discuss rename apiService to alfrescoApiService? it helps consistency

File: lib/content-services/document-list/components/library-name-column/library-name-column.component.ts [hunk 9]
```diff
@@ -16,75 +16,105 @@
...
+    @Input()
+    context: any;
+
+    displayTooltip$ = new BehaviorSubject<string>('');
+    displayText$ = new BehaviorSubject<string>('');
+    node: NodeEntry;
+
+    private sub: Subscription;
+
+    constructor(private element: ElementRef, private api: AlfrescoApiService) {}
```
Comment (eromano) [inline lib/content-services/document-list/components/library-name-column/library-name-column.component.ts]: would it better as we discuss rename api to alfrescoApiService? it helps consistency

File: lib/content-services/document-list/components/library-name-column/library-name-column.component.ts [hunk 10]
```diff
@@ -16,75 +16,105 @@
...
+            new CustomEvent('name-click', {
+                bubbles: true,
+                detail: {
+                    node: this.node
+                }
+            })
+        );
+    }
+
+    makeLibraryTooltip(library: any): string {
```
Comment (eromano) [inline lib/content-services/document-list/components/library-name-column/library-name-column.component.ts]: is it real any the type?

**Comment by DenysVuika** [comment]
@eromano I agree with all comments and will update the code after holidays 😊

File: lib/content-services/document-list/components/library-role-column/library-role-column.component.ts (hunk 2 above)
Comment (DenysVuika) [reply lib/content-services/document-list/components/library-role-column/library-role-column.component.ts]: Updated to use enum from js-api

File: lib/content-services/document-list/components/library-status-column/library-status-column.component.ts (hunk 3 above)
Comment (DenysVuika) [reply lib/content-services/document-list/components/library-status-column/library-status-column.component.ts]: using now enum from js-api

File: lib/core/datatable/components/datatable/date-cell.component.ts (hunk 6 above)
Comment (DenysVuika) [reply lib/core/datatable/components/datatable/date-cell.component.ts]: updated name

File: lib/content-services/document-list/components/library-name-column/library-name-column.component.ts (hunk 10 above)
Comment (DenysVuika) [reply lib/content-services/document-list/components/library-name-column/library-name-column.component.ts]: using the real type now

**Comment by eromano** [comment]
@DenysVuika  there are some e2e failing do you think is related to this modify? at the moment they are not failing on development

**Pull request merged**